src/tests/api/judging.test.ts

```diff
@@ -0,0 +1,145 @@
+import 'jest';
+import supertest from 'supertest';
+import { Team } from '../../entities/team';
+import { Judge } from '../../entities/judge';
+import { ExpandedTeamResult, bonusPointsForIdeaPitch } from '../../api/judging';
+import { createJudgeData, createTeamData, visitTeamsAndJudge } from '../utilities';
+import { createDbConnection, closedbConnection } from '../testdb';
+import { SupportRequest, SupportRequestType, SupportRequestStatus } from '../../entities/supportRequest';
+import { JudgingVote } from '../../entities/judgingVote';
+
+const adminSecret = 'Secrets are secretive';
+
+/* eslint-disable @typescript-eslint/no-var-requires, global-require */
+
+describe('api/judging', () => {
+  let teams: Team[];
+  let judges: Judge[];
+
+  beforeAll(async () => {
+    await createDbConnection();
+    teams = await createTeamData(10);
+    judges = await createJudgeData(10);
+    await visitTeamsAndJudge(judges, teams, 1);
+  });
+
+  afterAll(async () => {
+    await closedbConnection();
+  });
+
+  beforeEach(() => {
+    process.env.SLACK_SIGNING_SECRET = 'A JUNK TOKEN';
+    process.env.SLACK_BOT_TOKEN = 'ANOTHER JUNK TOKEN';
+    process.env.ADMIN_SECRET = adminSecret;
+  });
+
+  it('is protected by admin middleware', (done) => {
+    // Hide error output for unauth'd request
+    jest.mock('../../logger');
+
+    const { app } = require('../../app');
+    supertest(app)
+      .get('/api/judging/results/')
+      .set({
+        Authorization: "YOU DIDN'T SAY THE MAGIC WORD!",
+      })
+      .expect(401, done);
+  });
+
+  it('responds with expanded judging results', async () => {
+    const { app } = require('../../app');
+    const response = await supertest(app)
+      .get('/api/judging/results/')
+      .set({
+        Authorization: adminSecret,
+      })
+      .expect(200);
+
+    const results: ExpandedTeamResult[] = response.body;
+    expect(results).toBeDefined();
+  });
```

Comment:
Isn't this just testing that something got returned? Not that it actually got expanded judging results, since Typescript gets removed at compilation?

Reply:
That’s exactly what’s going on... is there another way to test it?? I couldn’t find anything as casting will succeed since this interface extends the other

Reply:
I meant shouldn’t we test what is actually returned? Like do an `.isEqual` to the object we expect?